Automattic/wp-calypso
# Framework: remove sitesList dependency from analytics

Currently when we fire a tracks event, properties like `blog_id` and `blog_lang` are populated using SitesList. This PR removes the SitesList dependency from the library and introduces a new middleware so analytics can have access to this data. Due to the number of events that might trigger a user/site update, I've opted to pass through dispatch to this library. This allows analytics to fire a custom event that the middleware can handle.

Let me know your thoughts on this approach.

![screen shot 2017-04-13 at 3 54 36 pm](https://cloud.githubusercontent.com/assets/1270189/25027461/a01008aa-2061-11e7-96e8-f8dfc9621beb.png)


### Testing Instructions
- in console, set debug to `localstorage.debug = 'calypso:analytics:tracks'`
- navigate to http://calypso.localhost:3000 events fire normally
- Click on my-sites, you should see a in redux devtools
![screen shot 2017-04-13 at 3 59 08 pm](https://cloud.githubusercontent.com/assets/1270189/25027535/2694e670-2062-11e7-83bb-202b073d83ff.png)
- tracks events should have the correct super props like `blog_id` and `blog_lang`
- switch to another site
- tracks events should have the new super props like `blog_id` and `blog_lang`
- in console calling `dispatch( { type: 'ANALYTICS_SUPER_PROPS_UPDATE' } )` will trigger the new middleware.
- other events (mc/google) are recorded normally.
